P2450 C++ Standard Library Issues to be moved in Virtual Plenary, Oct. 2021 (#5027)
cplusplus/draft: source/ranges.tex

\begin{itemdescr}
\pnum
Let \tcode{U} be \tcode{remove_reference_t<T>}
if \tcode{T} is an rvalue reference type, and \tcode{T} otherwise.
Given a variable \tcode{u} of type \tcode{U},
\tcode{T} models \libconcept{borrowed_range} only if

> the validity of iterators obtained from \tcode{U}

Reviewer: ```suggestion
the validity of iterators obtained from \tcode{u}
```

\item
Otherwise, if \tcode{T} models
\libconcept{random_access_range} and \libconcept{sized_range}
and is a specialization of
\tcode{span}\iref{views.span},
\tcode{basic_string_view}\iref{string.view}, or
\tcode{ranges::subrange}\iref{range.subrange},
then
\tcode{U(ranges::begin(E),
ranges::be\-gin(E) + std::min<D>(ranges::distance(E), F))},
except that \tcode{E} is evaluated only once,
where \tcode{U} is a type determined as follows:

\begin{itemize}
\item if \tcode{T} is a specialization of \tcode{span},
then \tcode{U} is \tcode{span<typename T::element_type>};
\item otherwise, if \tcode{T} is a specialization of \tcode{basic_string_view},
then \tcode{U} is \tcode{T};
\item otherwise, \tcode{T} is a specialization of \tcode{ranges::subrange}, and
\tcode{U} is \tcode{ranges::subrange<iterator_t<T>>};
\end{itemize}

\item
otherwise, if \tcode{T} is
a specialization of \tcode{ranges::iota_view}\iref{range.iota.view}
that models \libconcept{random_access_range} and \libconcept{sized_range},
then
\tcode{ranges::iota_view(*ranges::begin(E),
*(ranges::begin(E) + std::\linebreak{}min<D>(ranges::distance(E), F)))},

> except that \tcode{E} and \tcode{F} are each evaluated only once;

Reviewer: `F` is only mentioned once in the expression, so clearly it's evaluated only once and we don't need to say it again. I think this was a copy-paste error of mine, sorry :(

```suggestion
except that \tcode{E} is evaluated only once;
```
